pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -617,18 +615,29 @@ func (r *ReconcileDeadmansSnitchIntegration) deleteDMSClusterDeployment(dmsi *de
 }
 
 func instancesAreRunning(cd hivev1.ClusterDeployment) bool {
+	// Get hibernation PowerState a new ClusterDeployment Status field indicating if the cluster is running
+	// ie. The cluster is not "Resuming" if the PowerState is "Running", the cluster is operational.
+	// If the field is blank we move on and check the legacy reasons (It may be blank if the running version of
+	// Hive on cluster doesn't yet support it)
+	if cd.Status.PowerState != "" && cd.Status.PowerState == "Running" {
```

Comment:
The empty string check is redundant.
```suggestion
	if cd.Status.PowerState == "Running" {
```

Reply:
Checking !=  nil might avoid panics imo

Reply:
I don't think the compiler will even let you check for `nil` on a string type.

Reply:
ah right !

Reply:
Yeap :)